src/elfpy/pricing_models/yieldspace.py

```diff
@@ -353,12 +353,12 @@ def calc_in_given_out(
         # We precompute the YieldSpace constant k using the current reserves and
         # share price:
         #
-        # k = (c / μ) * (μ * z)**(1 - t) + (2y + cz)**(1 - t)
+        # k = (c / μ) * (μ * z)**(1 - tau) + (2y + cz)**(1 - tau)
         k = price_utils.calc_k_const(market_state, time_elapsed)
         if out.unit == TokenType.BASE:
             in_reserves = market_state.bond_reserves + total_reserves
             out_reserves = market_state.share_reserves
-            d_shares = out.amount / market_state.share_price
+            d_shares = out.amount
```

Comment:
Out is denominated in base not shares (just like how our smart contract will accept base rather than shares). The previous implementation was correct.

Reply:
Ok look at this one again.  The next line of code, line 371 has:
```
              without_fee_or_slippage = (1 / spot_price) * market_state.share_price * d_shares
 ```
 
 since the output is base, we multiply by c since `x = c * z`

Reply:
Let's chat about this after all-hands.